src/Microsoft.AspNetCore.Mvc.ViewFeatures/Internal/ViewBufferTextWriter.cs

```diff
@@ -1,211 +1,436 @@
-﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
 
 using System;
-using System.Buffers;
-using System.Collections.Generic;
-using System.Diagnostics;
 using System.IO;
 using System.Text;
+using System.Text.Encodings.Web;
 using System.Threading.Tasks;
+using Microsoft.AspNetCore.Html;
+using Microsoft.AspNetCore.Mvc.Internal;
 
 namespace Microsoft.AspNetCore.Mvc.ViewFeatures.Internal
 {
+    /// <summary>
+    /// <para>
+    /// A <see cref="TextWriter"/> that is backed by a unbuffered writer (over the Response stream) and/or a 
+    /// <see cref="ViewBuffer"/>
+    /// </para>
+    /// <para>
+    /// When <c>Flush</c> or <c>FlushAsync</c> is invoked, the writer copies all content from the buffer to
+    /// the writer and switches to writing to the unbuffered writer for all further write operations.
+    /// </para>
+    /// </summary>
     public class ViewBufferTextWriter : TextWriter
```

Comment:
Combines the best features of `RazorTextWriter` and `HtmlContentWrapperTextWriter`